### PR TITLE
Task/morecast style

### LINKED ISSUE #3371: MoreCast 2.0: Implement Styling Updates to MoreCast tables
**As a** ForeCaster
**I need the** prioritized items (below) from the Jan UX Forecaster workshop implemented
**So that I** can validate whether the design suggestions are effective for my accessibility needs.

**Acceptance Criteria**
  - [x] Columns, cells, headers, etc. grouped and differentiated by colour as per UX design
  - [ ] Cells need to have enough room to show 5 characters (ex: -22.8) without being cut off.
  

**Additional Context**
- As an outcome of early January's Journey Mapping session with the Forecasters, accessibility issues were raised regarding working with the MoreCast tables, as listed above in the AC. Outcomes of the workshop were captured in [Miro board and prioritized](https://miro.com/app/board/uXjVN7wCOfc=/?moveToWidget=3458764575119977399&cot=14).
- https://www.figma.com/file/0iKI82NZZtYR8UrdTMdVLD/MoreCast-2.0?type=design&node-id=337-1908&mode=design&t=UrXGPWxoU0j6tQUA-4
- related to #3394

### PR DESCRIPTION
**Pull request by conbrad**
- Overrides `MuiInputBase` globally to have a font size of `1em` for consistency with FireBAT and for better accommodation of values in Morecast2
- Adds color coding of weather parameters and weather models to headers, columns and cells of Morecast 2
- closes #3371 

# Test Links:
[Landing Page](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/)
[MoreCast 2.0](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/morecast-2)
[Percentile Calculator](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/percentile-calculator)
[MoreCast](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN,s=9999&f=c7&c=NaN&w=NaN)
[Auto Spatial Advisory (ASA)](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/auto-spatial-advisory)
[HFI Calculator](https://wps-pr-3422.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

File: web/src/features/moreCast2/components/DataGridColumns.tsx [hunk 1]
```diff
@@ -111,7 +112,7 @@ export class DataGridColumns {
 
 const renderGroupHeader = (
   id: string,
-  weatherParam: string,
+  weatherParam: keyof MoreCastParamColors,
```
Comment (dgboss) [inline web/src/features/moreCast2/components/DataGridColumns.tsx:115]: The `weatherParam` variable is used for more than just color, so I'm not sure it makes sense to relate it to color here?
I messed up by not typing this properly when I did the show hide columns work.

File: web/src/features/moreCast2/components/ForecastDataGrid.tsx [hunk 2]
```diff
@@ -15,21 +15,50 @@ import { LinearProgress } from '@mui/material'
...
+    styles[`& .${key}-header`] = {
+      backgroundColor: MORECAST_MODEL_COLORS[key as keyof MoreCastModelColors].bg,
+      borderBottom: 'solid',
+      borderRight: 'solid',
+      // Ugly override, tried to avoid, but MUI overwrites border with it's own otherwise
+      borderColor: `${MORECAST_MODEL_COLORS[key as keyof MoreCastModelColors].border} !important`
+    }
+  })
+
+  return styles
```
Comment (dgboss) [inline web/src/features/moreCast2/components/ForecastDataGrid.tsx:60]: Nice! I didn't realize you could do this!

File: web/src/features/moreCast2/components/GroupHeader.tsx [hunk 3]
```diff
@@ -1,14 +1,14 @@
...
+import { LIGHT_GREY, MEDIUM_GREY, DARK_GREY, MoreCastParamColors } from 'app/theme'
 import { ColumnVis } from 'features/moreCast2/components/DataGridColumns'
 
 interface GroupHeaderProps {
   id: string
   columns: ColumnVis[]
-  weatherParam: string
-  handleShowHideChange: (weatherParam: string, columnName: string, value: boolean) => void
+  weatherParam: keyof MoreCastParamColors
+  handleShowHideChange: (weatherParam: keyof MoreCastParamColors, columnName: string, value: boolean) => void
```
Comment (dgboss) [inline web/src/features/moreCast2/components/GroupHeader.tsx:11]: We should update the type of the `weatherParam` parameter in `handleShowHideChange()` in `TabbedDataGrid` as well.

**Review by dgboss — APPROVED** [review]
Strong work! I learned a few things!

File: web/src/features/moreCast2/components/ForecastDataGrid.tsx (hunk 2 above)
Comment (conbrad) [reply web/src/features/moreCast2/components/ForecastDataGrid.tsx:60]: Me neither, I learned this from @brettedw when I took over the PR last week

Comment (brettedw) [reply web/src/features/moreCast2/components/ForecastDataGrid.tsx:60]: Can't take all the credit, found an example of someone doing something similar 🙂

File: web/src/features/moreCast2/components/SelectableButton.tsx [hunk 4]
```diff
@@ -1,25 +1,58 @@
 import React from 'react'
 
 import { Button } from '@mui/material'
-import { theme } from 'app/theme'
+import { MORECAST_WEATHER_PARAM_COLORS, MoreCastParamColors, theme } from 'app/theme'
+import styled from '@emotion/styled'
+
+const StyledButton = styled(Button)(() => ({
+  marginLeft: theme.spacing(1),
+  '&:hover': {
```
Comment (brettedw) [inline web/src/features/moreCast2/components/SelectableButton.tsx:9]: Not code related but I made a comment on Figma that Raj hasn't replied to yet. Maybe it's minor but curious what you think.
When hover is white, it's hard to tell when you've actually clicked on the tab. You have to move your cursor off of the button to see if it's changed colour to active/inactive.
Take it or leave it, just a thought

**Review by brettedw — COMMENTED** [review]
Looks good! I do see where Darren is coming from on the `weatherParam` typing.

Only other thought is the grey backgroundColor Raj has on the Actual/Forecast columns on Figma. Maybe you guys had a convo about that already though

File: web/src/features/moreCast2/components/DataGridColumns.tsx (hunk 1 above)
Comment (conbrad) [reply web/src/features/moreCast2/components/DataGridColumns.tsx]: Renamed to `MoreCastParams`

File: web/src/features/moreCast2/components/SelectableButton.tsx (hunk 4 above)
Comment (conbrad) [reply web/src/features/moreCast2/components/SelectableButton.tsx:9]: Chatted about it and got the green light

**Review by brettedw — APPROVED** [review]
Looks great!

**Review by dgboss — APPROVED** [review]
Looks great!

**Pull request closed without merging**